mysql/resource_grant.go

```diff
@@ -578,25 +598,37 @@ func showUserGrants(ctx context.Context, db *sql.DB, user string) ([]*MySQLGrant
 			continue
 		}
 
-		if m := re.FindStringSubmatch(rawGrant); len(m) == 5 {
+		if m := re.FindStringSubmatch(rawGrant); len(m) == 6 {
 			privsStr := m[1]
 			privList := extractPermTypes(privsStr)
 			privileges := make([]string, len(privList))
 
 			for i, priv := range privList {
 				privileges[i] = strings.TrimSpace(priv)
 			}
-			grantUserHost := m[4]
+			grantUserHost := m[5]
 			if normalizeUserHost(grantUserHost) != normalizeUserHost(user) {
 				// Percona returns also grants for % if we requested IP.
 				// Skip them as we don't want terraform to consider it.
 				log.Printf("[DEBUG] Skipping grant with host %v while we want %v", grantUserHost, user)
 				continue
 			}
 
+			// A legacy feature of the Hashicorp MySQL provider is to allow
+			// procedures to be specified as a database. This logic supports that
+			// legacy feature.
+			var database, table string
+			if m[2] == "" {
+				database = strings.Trim(m[3], "`\"")
+				table = strings.Trim(m[4], "`\"")
+			} else {
+				database = fmt.Sprintf("%s%s.%s", m[2], strings.Trim(m[3], "`\""), strings.Trim(m[4], "`\""))
```

Comment:
Please consider adding a comment the expected value. On a first sight, I had absolutely no idea about what should be saved to "database" and why.

Reply:
Did you do anything here? Please at least leave a comment if you want to resolve something.

Reply:
Sorry, I am actively working on this. Just haven't pushed my latest changes.

Reply:
Ok thanks! Feel free to push them; more changes can also be added later.